admin/create-theme/theme-templates.php

```diff
@@ -103,88 +124,219 @@ public static function clear_user_templates_customizations() {
+		//TODO: This seems messy.  How can we make this more elegant?
+		$content = $block['innerContent'][0];
+		$doc     = new DOMDocument();
+		$doc->loadHTML( $content, LIBXML_HTML_NOIMPLIED | LIBXML_HTML_NODEFDTD );
+		$elements = $doc->getElementsByTagName( '*' );
+		foreach ( $elements as $element ) {
+			// phpcs:ignore
+			$element->nodeValue = self::escape_text( $element->nodeValue );
+		}
+		$block['innerContent'][0] = html_entity_decode( $doc->saveHTML() );
+		return $block;
+	}
 
-				// If there are external images, add it as a pattern
-				$pattern                 = Theme_Patterns::pattern_from_template( $template_data );
-				$pattern_link_attributes = array(
-					'slug' => $pattern['slug'],
-				);
-				$template_data->content  = Theme_Patterns::create_pattern_link( $pattern_link_attributes );
+	public static function escape_text( $text ) {
+		if ( ! $text ) {
+			return $text;
+		}
+		return "<?php echo __('" . $text . "', '" . wp_get_theme()->get( 'TextDomain' ) . "');?>";
+	}
 
-				// Write the pattern
-				file_put_contents(
-					get_stylesheet_directory() . DIRECTORY_SEPARATOR . 'patterns' . DIRECTORY_SEPARATOR . $template_data->slug . '.php',
-					$pattern['content']
-				);
+	public static function eliminate_environment_specific_content( $template ) {
+
+		$template_blocks = parse_blocks( $template->content );
+		$blocks          = _flatten_blocks( $template_blocks );
+
+		foreach ( $blocks as $key => $block ) {
+
+			// remove theme attribute from template parts
+			if ( 'core/template-part' === $block['blockName'] && isset( $block['attrs']['theme'] ) ) {
+				unset( $blocks[ $key ]['attrs']['theme'] );
 			}
 
-			// Write the template content
-			file_put_contents(
-				get_stylesheet_directory() . DIRECTORY_SEPARATOR . $template_folders['wp_template_part'] . DIRECTORY_SEPARATOR . $template_data->slug . '.html',
-				$template_data->content
-			);
+			// remove ref attribute from blocks
+			// TODO: are there any other blocks that have refs?
+			if ( 'core/navigation' === $block['blockName'] && isset( $block['attrs']['ref'] ) ) {
+				unset( $blocks[ $key ]['attrs']['ref'] );
+			}
+
+			if ( in_array( $block['blockName'], array( 'core/image', 'core/cover' ), true ) ) {
+				// remove id attribute from image and cover blocks
+				// TODO: are there any other blocks that have ids?
+				if ( isset( $block['attrs']['id'] ) ) {
+					unset( $blocks[ $key ]['attrs']['id'] );
+				}
+
+				// remove wp-image-[id] class from image and cover blocks
+				if ( isset( $block['attrs']['className'] ) ) {
+					$blocks[ $key ]['attrs']['className'] = preg_replace( '/wp-image-\d+/', '', $block['attrs']['className'] );
+				}
+			}
+
+			// set taxQuery to null for query blocks
+			if ( 'core/query' === $block['blockName'] ) {
+				if ( isset( $block['attrs']['taxQuery'] ) ) {
+					unset( $blocks[ $key ]['attrs']['taxQuery'] );
+				}
+				if ( isset( $block['attrs']['queryId'] ) ) {
```

Comment:
We would need to preserve `queryId` for [this reason](https://github.com/WordPress/wordpress-develop/pull/3899#discussion_r1104638014).

Reply:
I removed that bit.